File: jianzhioffer/twostackforqueue.py
# -*- coding:utf-8 -*-
class Solution:
    def __init__(self):
        self.stack1 = []
        self.stack2 = []
        
    def push(self, node):
        # write code here
        self.stack1.append(node)
    def pop(self):
        # return xx
        if self.stack2:
            return self.stack2.pop()
        elif not self.stack1:
            return None
        else:
            while self.stack1:
                self.stack2.append(self.stack1.pop())
            return self.stack2.pop()
def test_pop_push():
    solve = Solution()
    for i in xrange(1, 10):
        solve.push(i)
    print solve.pop()
    print solve.pop()

if __name__ == '__main__':
    test_pop_push()

File: jianzhioffer/twostackforqueue.cpp
#include <stack>
using namespace std;

class Solution
{
public:
    void push(int node) 
    {
    	this->stack1.push(node);    
    }

    int pop() 
    {
    	if(!this->stack2.empty())
    	{
    		int result = this->stack2.top();
    		this->stack2.pop();
    		return result;
    	}
    	else if(this->stack1.empty())
    	{
    		return NULL;
    	}
    	else
    	{
    		while(!this->stack1.empty())
    		{
    			this->stack2.push(this->stack1.top());
    			this->stack1.pop();
    		}
    		int result = this->stack2.top();
    		this->stack2.pop();
    		return result;
    	}
    }

private:
    stack<int> stack1;
    stack<int> stack2;
};